jaeger-core/src/main/java/com/uber/jaeger/metrics/InMemoryMetricsFactory.java

```diff
@@ -0,0 +1,175 @@
+   * @param tags the metric tags
+   * @return the counter value or -1, if no counter exists for the given metric name
+   */
+  public long getCounter(String name, Map<String, String> tags) {
+    return getValue(counters, name, tags);
+  }
+
+  /**
+   * Returns the current value for the gauge with the given metric name. Note that the metric name is not the gauge
+   * name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags as comma separated list of entries, like "foo=bar,baz=qux"
+   * @return the gauge value or -1, if no gauge exists for the given metric name
+   */
+  public long getGauge(String name, String tags) {
+    return getValue(gauges, name, tags);
+  }
+
+  /**
+   * Returns the current value for the gauge with the given metric name. Note that the metric name is not the gauge
+   * name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags
+   * @return the gauge value or -1, if no gauge exists for the given metric name
+   */
+  public long getGauge(String name, Map<String, String> tags) {
+    return getValue(gauges, name, tags);
+  }
+
+  /**
+   * Returns the current accumulated timing information for the timer with the given metric name.
+   * Note that the metric name is not the timer name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags as comma separated list of entries, like "foo=bar,baz=qux"
+   * @return the timer value or -1, if no timer exists for the given metric name
+   */
+  public long getTimer(String name, String tags) {
+    return getValue(timers, name, tags);
+  }
+
+  /**
+   * Returns the current accumulated timing information for the timer with the given metric name.
+   * Note that the metric name is not the timer name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags
+   * @return the timer value or -1, if no timer exists for the given metric name
+   */
+  public long getTimer(String name, Map<String, String> tags) {
+    return getValue(timers, name, tags);
+  }
+
+  private long getValue(Map<String, AtomicLong> collection, String name, String tags) {
+    if (null == tags || tags.isEmpty()) {
+      return getValue(collection, name);
+    }
+
+    String[] entries = tags.split(",");
+    Map<String, String> tagsAsMap = new HashMap<String, String>(entries.length);
+    for (String entry : entries) {
+      String[] keyValue = entry.split("=");
+      tagsAsMap.put(keyValue[0], keyValue[1]);
```

Comment:
There should be a check otherwise it will throw `IndexOutOfBoundsException`

Reply:
I don't think it would matter that much, but I added this check anyway